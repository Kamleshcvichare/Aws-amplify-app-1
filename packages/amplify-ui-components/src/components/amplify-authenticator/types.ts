export enum AuthState {
  LoggedOut = 'LoggedOut',
  LoggedIn = 'LoggedIn',
}

export type UserData = {
  username?: string;
};

export type Creds = {
  username?: string;
  password?: string;
};
