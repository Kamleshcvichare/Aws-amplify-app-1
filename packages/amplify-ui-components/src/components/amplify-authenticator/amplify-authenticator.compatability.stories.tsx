/* @jsx createElement */

import * as knobs from '@storybook/addon-knobs';
import { Authenticator } from 'aws-amplify-react';
import { createElement, StrictMode } from 'react';
import { render } from 'react-dom';

export default {
  title: 'amplify-authenticator/compatability',
  decorators: [
    story => {
      const node = document.createElement('main');
      render(<StrictMode>{story()}</StrictMode>, node);
      return node;
    },
  ],
};

export const defaults = () => <Authenticator />;

export const withFederated = () => {
  return (
    <Authenticator
      federated={{
        amazon_client_id: knobs.text('Amazon client ID', 'amazon_client_id'),
        auth0: {
          clientID: knobs.text('Auth0 client ID', 'auth0_client_id'),
          domain: knobs.text('Auth0 account domain', 'example.auth0.com'),
        },
        facebook_app_id: knobs.text('Facebook app ID', 'facebook_app_id'),
        google_client_id: knobs.text('Google client ID', 'google_client_id'),
        oauth_config: {},
      }}
    />
  );
};
