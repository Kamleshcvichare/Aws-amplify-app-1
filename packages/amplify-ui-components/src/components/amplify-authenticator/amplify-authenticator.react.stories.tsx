/** @jsx createElement */

import * as knobs from '@storybook/addon-knobs';
import { createElement } from 'react';
import { render } from 'react-dom';

import { AmplifyAuthenticator } from '../../../../amplify-ui-react';

console.log({ AmplifyAuthenticator });

export default {
  title: 'amplify-authenticator/react',
  decorators: [
    story => {
      const root = document.createElement('main');

      render(story(), root);

      return root;
    },
  ],
};

export const defaults = () => <AmplifyAuthenticator />;

export const withFederated = () => (
  <AmplifyAuthenticator
    federated={{
      amazonClientId: knobs.text('Amazon client ID', 'amazon_client_id'),
      auth0Config: {
        clientID: knobs.text('Auth0 client ID', 'auth0_client_id'),
        domain: knobs.text('Auth0 account domain', 'example.auth0.com'),
        redirectUri: 'http://localhost:3000/',
        responseType: 'token id_token',
      },
      facebookAppId: knobs.text('Facebook app ID', 'facebook_app_id'),
      googleClientId: knobs.text('Google client ID', 'google_client_id'),
      oauthConfig: {},
    }}
  />
);
