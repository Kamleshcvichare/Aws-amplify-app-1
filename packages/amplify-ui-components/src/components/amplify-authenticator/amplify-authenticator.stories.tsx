import { storiesOf } from '@storybook/html';

const authenticatorStories = storiesOf('amplify-authenticator', module);

authenticatorStories.add('default', () => {
  return `<amplify-authenticator></amplify-authenticator>`;
});
