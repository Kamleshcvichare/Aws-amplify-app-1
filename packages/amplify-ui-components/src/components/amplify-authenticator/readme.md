# amplify-authenticator



<!-- Auto Generated Below -->


## Properties

| Property           | Attribute            | Description                                                                                                                     | Type                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | Default            |
| ------------------ | -------------------- | ------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------ |
| `initialAuthState` | `initial-auth-state` | Initial starting state of the Authenticator component. E.g. If `signup` is passed the default component is set to AmplifySignUp | `AuthState.ConfirmSignIn \| AuthState.ConfirmSignUp \| AuthState.CustomConfirmSignIn \| AuthState.ForgotPassword \| AuthState.Loading \| AuthState.ResetPassword \| AuthState.SettingMFA \| AuthState.SignIn \| AuthState.SignOut \| AuthState.SignUp \| AuthState.SignedIn \| AuthState.SigningUp \| AuthState.TOTPSetup \| AuthState.VerifyContact \| AuthState.VerifyingAttributes \| AuthState.confirmingSignInCustomFlow \| AuthState.confirmingSignUpCustomFlow \| AuthState.signedOut` | `AuthState.SignIn` |


## Dependencies

### Used by

 - [amplify-examples](../amplify-examples)

### Depends on

- [amplify-sign-in](../amplify-sign-in)
- [amplify-sign-up](../amplify-sign-up)
- [amplify-confirm-sign-up](../amplify-confirm-sign-up)
- [amplify-forgot-password](../amplify-forgot-password)
- context-consumer

### Graph
```mermaid
graph TD;
  amplify-authenticator --> amplify-sign-in
  amplify-authenticator --> amplify-sign-up
  amplify-authenticator --> amplify-confirm-sign-up
  amplify-authenticator --> amplify-forgot-password
  amplify-authenticator --> context-consumer
  amplify-sign-in --> amplify-link
  amplify-sign-in --> amplify-form-section
  amplify-sign-in --> amplify-auth-fields
  amplify-form-section --> amplify-button
  amplify-form-section --> amplify-section
  amplify-auth-fields --> amplify-username-field
  amplify-auth-fields --> amplify-password-field
  amplify-auth-fields --> amplify-email-field
  amplify-auth-fields --> amplify-code-field
  amplify-auth-fields --> amplify-form-field
  amplify-auth-fields --> amplify-form-field
  amplify-username-field --> amplify-form-field
  amplify-form-field --> amplify-label
  amplify-form-field --> amplify-input
  amplify-form-field --> amplify-hint
  amplify-password-field --> amplify-form-field
  amplify-email-field --> amplify-form-field
  amplify-code-field --> amplify-form-field
  amplify-sign-up --> amplify-form-section
  amplify-sign-up --> amplify-auth-fields
  amplify-sign-up --> amplify-link
  amplify-sign-up --> amplify-button
  amplify-confirm-sign-up --> amplify-link
  amplify-confirm-sign-up --> amplify-form-section
  amplify-confirm-sign-up --> amplify-auth-fields
  amplify-forgot-password --> amplify-form-section
  amplify-forgot-password --> amplify-link
  amplify-forgot-password --> amplify-auth-fields
  amplify-examples --> amplify-authenticator
  style amplify-authenticator fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
