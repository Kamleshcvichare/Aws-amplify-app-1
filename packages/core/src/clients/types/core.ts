/**
 * General None HTTP-specific request interface
 */
export interface Request {
	url: URL;
	body?: unknown;
}

export interface Response {
	body: unknown;
}

export interface TransferHandler<
	Input extends Request,
	Output extends Response,
	TransferOptions
> {
	(request: Input, options: TransferOptions): Promise<Output>;
}

/**
 * A slimmed down version of the AWS SDK v3 middleware handler, only handling instantiated requests
 */
export type MiddlewareHandler<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (request: Input, options: MiddlewareOptions) => Promise<Output>;

export type MiddlewareContext = Record<string, unknown>;

/**
 * A slimmed down version of the AWS SDK v3 middleware, only handling tasks after Serde.
 */
export type Middleware<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (
	next: MiddlewareHandler<Input, Output, MiddlewareOptions>,
	context: MiddlewareContext
) => MiddlewareHandler<Input, Output, MiddlewareOptions>;
