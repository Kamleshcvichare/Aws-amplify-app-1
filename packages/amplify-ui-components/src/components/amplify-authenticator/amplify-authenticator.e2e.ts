import { newE2EPage } from '@stencil/core/testing';
import { pixelThreshold } from '../../common/testing';

describe('amplify-authenticator e2e:', () => {
  it('should render a amplify-authenticator', async () => {
    const page = await newE2EPage();
    await page.setContent(`<amplify-authenticator></amplify-authenticator>`);
    const el = await page.find('amplify-authenticator');
    expect(el).not.toBeNull();

    const screenshot = await page.compareScreenshot('amplify-authenticator', { fullPage: true });
    expect(screenshot).toMatchScreenshot({ allowableMismatchedPixels: pixelThreshold });
  });
});