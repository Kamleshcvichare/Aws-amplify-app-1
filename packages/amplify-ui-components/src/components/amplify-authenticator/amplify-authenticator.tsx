import { Component, State, Prop, h } from '@stencil/core';
import { AuthState } from '../../common/types/auth-types';
import { AuthStateTunnel } from '../../data/auth-state';

@Component({
  tag: 'amplify-authenticator',
  shadow: false,
})
export class AmplifyAuthenticator {
  /** Initial starting state of the Authenticator component. E.g. If `signup` is passed the default component is set to AmplifySignUp */
  @Prop() initialAuthState: AuthState = AuthState.SignIn;
  /** Used as a flag in order to trigger the content displayed */
  @State() authState: AuthState = AuthState.Loading;

  componentWillLoad() {
    this.authState = this.initialAuthState;
  }

  onAuthStateChange = (nextAuthState: AuthState, data?: object) => {
    console.log(data);
    if (nextAuthState === undefined) return console.info('nextAuthState cannot be undefined');

    console.info('Inside onAuthStateChange Method current authState:', this.authState);
    this.authState = nextAuthState;
    console.info(`authState has been updated to ${this.authState}`);
  };

  renderAuthComponent(authState: AuthState) {
    switch (authState) {
      case AuthState.Loading:
        return <div>Loading...</div>;
      case AuthState.SignIn:
        return <amplify-sign-in handleAuthStateChange={this.onAuthStateChange} />;
      case AuthState.SignOut:
        // TODO: add sign out component
        return <div>Sign Out Component</div>;
      case AuthState.SignUp:
        return <amplify-sign-up handleAuthStateChange={this.onAuthStateChange} />;
      case AuthState.ConfirmSignUp:
        // TODO: add confirm sign up component
        return <div>Confirm Sign Up Component</div>;
      case AuthState.ForgotPassword:
        return <amplify-forgot-password handleAuthStateChange={this.onAuthStateChange} />;
      case AuthState.ResetPassword:
        // TODO: add forgot password component
        return <div>Reset Password Component</div>;
    }
  }

  render() {
    const tunnelState = {
      authState: this.authState,
      onAuthStateChange: this.onAuthStateChange,
    };
    return (
      <AuthStateTunnel.Provider state={tunnelState}>
        {this.renderAuthComponent(this.authState)}
      </AuthStateTunnel.Provider>
    );
  }
}
