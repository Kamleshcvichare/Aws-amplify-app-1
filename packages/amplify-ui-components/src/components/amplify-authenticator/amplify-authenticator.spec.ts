import { newSpecPage } from '@stencil/core/testing';
import { AmplifyAuthenticator } from './amplify-authenticator';

describe('amplify-authenticator spec:', () => {
  describe('Component logic ->', () => {
    let authenticator;

    beforeEach(() => {
      authenticator = new AmplifyAuthenticator();
    });

    it('should render `firstInitialLoad` to true by default', () => {
      expect(authenticator.firstInitialLoad).toBe(true);
    });

    it('should render `authState` as `loading` by default', () => {
      expect(authenticator.authState).toEqual('loading');
    });

    it('should render `state` as `loading` by default', () => {
      expect(authenticator.state).toEqual('loading');
    });
  });
  describe('Render logic ->', () => {
    it('should render a `sign up` by passing `signup` into state prop', async () => {
      const page = await newSpecPage({
        components: [AmplifyAuthenticator],
        html: `<amplify-authenticator state="signup"></amplify-authenticator>`
      });

      expect(page.root).toMatchSnapshot();
    });
    it('should render a `forgot password` by passing `forgotpassword` into state prop', async () => {
      const page = await newSpecPage({
        components: [AmplifyAuthenticator],
        html: `<amplify-authenticator state="forgotpassword"></amplify-authenticator>`
      });

      expect(page.root).toMatchSnapshot();
    });

    it('should render a `sign in` by default', async () => {
      const page = await newSpecPage({
        components: [AmplifyAuthenticator],
        html: `<amplify-authenticator></amplify-authenticator>`
      });

      expect(page.root).toMatchSnapshot();
    });
  });
});