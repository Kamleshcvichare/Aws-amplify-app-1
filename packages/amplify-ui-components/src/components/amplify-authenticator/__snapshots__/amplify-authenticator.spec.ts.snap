// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-authenticator spec: Render logic -> should render a \`forgot password\` if initialAuthState is \`forgotpassword\` 1`] = `
<amplify-authenticator initial-auth-state="forgotpassword">
  <div>
    Forgot Password Component
  </div>
</amplify-authenticator>
`;

exports[`amplify-authenticator spec: Render logic -> should render a \`sign in\` by default 1`] = `
<amplify-authenticator>
  <amplify-sign-in></amplify-sign-in>
</amplify-authenticator>
`;

exports[`amplify-authenticator spec: Render logic -> should render a \`sign up\` if initialAuthState is \`signup\` 1`] = `
<amplify-authenticator initial-auth-state="signup">
  <div>
    Sign Up Component
  </div>
</amplify-authenticator>
`;
